packages/amplify-graphql-model-transformer/src/resolvers/generator/vtl-generator.ts

```diff
@@ -0,0 +1,21 @@
+import { ModelDirectiveConfiguration } from "../../directive";
+
+export interface OperationConfig {
+  operation?: string;
+  operationName?: string;
+}
+
+export interface ModelVTLGenerator {
+  generateUpdateRequestTemplate(modelName: string, isSyncEnabled: boolean, config?: OperationConfig): string;
+  generateCreateRequestTemplate(modelName: string, modelIndexFields: string[], config?: OperationConfig): string;
+  generateCreateInitSlotTemplate(modelConfig: ModelDirectiveConfiguration, config?: OperationConfig): string;
+  generateDeleteRequestTemplate(modelName: string, isSyncEnabled: boolean, config?: OperationConfig): string;
+  generateUpdateInitSlotTemplate(modelConfig: ModelDirectiveConfiguration, config?: OperationConfig): string;
+  generateGetRequestTemplate(config?: OperationConfig): string;
+  generateGetResponseTemplate(isSyncEnabled: boolean, config?: OperationConfig): string;
+  generateListRequestTemplate(config?: OperationConfig): string;
+  generateSyncRequestTemplate(config?: OperationConfig): string;
+  generateSubscriptionRequestTemplate(config?: OperationConfig): string;
+  generateSubscriptionResponseTemplate(config?: OperationConfig): string;
+  generateDefaultResponseMappingTemplate(isSyncEnabled: boolean, mutation: boolean): string;
+}
```

Comment:
qq: can modelName, modelConfig, and isSyncEnabled be pulled out into a constructor? what is really needed to make a decision on a per-field basis? Also, why is the OperationConfig optional all the way down? When do we not know the operation and operationName

Reply:
100% agreed. This can be improved. I carried forward the existing argument list and added anything that is additionally required for RDS to `config` argument. But this can be improved something like the below, will take care of it in another PR (I'm still not sure how we can use a single type structure for all these methods as the argument varies for each generate* method).

```ts
type RDSModelConfig {
...
}

type DynamoDBModelConfig {
...
}

type ModelConfig = RDSModelConfig | DynamoDBModelConfig;
```

If single config object is not sufficient, then the best approach is to create a config type based for each operation like the below so that the methods are strongly typed.

generateUpdateRequestTemplate(config: ModelConfig): string;

```ts
type RDSModelCreateRequestConfig {
...
}

type DynamoDBModelCreateRequestConfig {
...
}

type ModelCreateRequestConfig = RDSModelCreateRequestConfig | DynamoDBModelCreateRequestConfig;

generateUpdateRequestTemplate(config: ModelCreateRequestConfig): string;

// Similarly create config types for other operations

```

Reply:
> I'm still not sure how we can use a single type structure for all these methods as the argument varies for each generate* method

I think that's not actually something we need. I'd prefer to have per-method strongly typed props, rather than a single `ConfigOptions` shape that doesn't guarantee correctness for individual methods.

Reply:
I have added a task to RDS project for this.